HDDS-4102. Normalize Keypath for lookupKey.

bharatviswa504 opened the pull request:
## What changes were proposed in this pull request?

When ozone.om.enable.filesystem.paths, OM normalizes path, and stores the Keyname.

Now when user tries to read the file from S3 using the keyName which user has used to create the Key, it will return error KEY_NOT_FOUND

The issue is, `lookupKey` also need to normalize path, when ozone.om.enable.filesystem.paths is enabled. This is a common API used by S3/FS.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-4102

## How was this patch tested?

Added a test.

rakeshadr commented:
Thanks @bharatviswa504  for the contribution. 

TestOzoneFileSystem#testDeleteCreatesFakeParentDir test is failing with this fix. Please take care this.

elek commented:
Thanks the patch @bharatviswa504. Cross-linking my generic questions: https://issues.apache.org/jira/browse/HDDS-4097

bharatviswa504 commented:
Mistakenly for this PR branch was created in apache repo.
Deleted my work in progress branch from apache repo, and created a new PR for this #1328 

@elek 
I have replied to your comments in Jira.

@rakeshadr 
Thanks for the review. Modified code to fix the test.

elek commented:
Let me clear: these are not blocking questions, just parallel conversations which is related.
